src/groups/general_linear.jl

```diff
@@ -88,12 +92,12 @@ end
 function exp!(G::GeneralLinear, q, p, X, t::Number)
     return exp!(G, q, p, t * X)
 end
-function exp!(::GeneralLinear{1}, q, p, X)
```

Comment:
This is again the problem of forwarding from type to field parameter implementation. That is possible but requires adding forwarding methods. In general things should work for field parameters but performance-wise, it's easier to require users to use type-parameters for parameter-specific optimizations rather than checking values of parameters in the generic method for possible faster implementations. But doing such checks and forwards is definitely possible.